system/Commands/Generators/MigrationGenerator.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+/**
+ * This file is part of the CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Commands\Generators;
+
+use CodeIgniter\CLI\BaseCommand;
+use CodeIgniter\CLI\CLI;
+use CodeIgniter\CLI\GeneratorTrait;
+
+/**
+ * Generates a skeleton migration file.
+ */
+class MigrationGenerator extends BaseCommand
+{
+	use GeneratorTrait;
+
+	/**
+	 * The Command's Group
+	 *
+	 * @var string
+	 */
+	protected $group = 'Generators';
+
+	/**
+	 * The Command's Name
+	 *
+	 * @var string
+	 */
+	protected $name = 'make:migration';
+
+	/**
+	 * The Command's Description
+	 *
+	 * @var string
+	 */
+	protected $description = 'Generates a new migration file.';
+
+	/**
+	 * The Command's Usage
+	 *
+	 * @var string
+	 */
+	protected $usage = 'make:migration <name> [options]';
+
+	/**
+	 * The Command's Arguments
+	 *
+	 * @var array
+	 */
+	protected $arguments = [
+		'name' => 'The migration class name.',
+	];
+
+	/**
+	 * The Command's Options
+	 *
+	 * @var array
+	 */
+	protected $options = [
+		'--session'   => 'Generates the migration file for database sessions.',
+		'--namespace' => 'Set root namespace. Defaults: "APP_NAMESPACE".',
+		'--suffix'    => 'Append the component title to the class name (e.g. UserMigration => User).',
+	];
+
+	/**
+	 * Actually execute a command.
+	 *
+	 * @param array $params
+	 */
+	public function run(array $params)
+	{
+		$this->component = 'Migration';
+		$this->directory = 'Database\Migrations';
+		$this->template  = 'migration.tpl.php';
+
+		if ($this->getOption('session'))
+		{
+			$params[0] = '_create_ci_session_table';
+		}
+
+		$this->execute($params);
+	}
+
+	/**
+	 * Prepare options and do the necessary replacements.
+	 *
+	 * @param string $class
+	 *
+	 * @return string
+	 */
+	protected function prepare(string $class): string
+	{
+		$data = [
+			'session' => false,
+			'matchIP' => false,
+		];
+
+		if ($this->getOption('session'))
+		{
+			$table = CLI::prompt(lang('CLI.generator.tableName')); // @codeCoverageIgnore
```

Comment:
This is the same problem as with `make:model`. Also on line 111.

Reply:
I think table name should be equal to model name, most of tables name aren't plural

Reply:
I guess thats kind of personal preference, but personally, I'd go with pluralized forms.
Imagine having a database for your car shop, then you would have a `cars` model and a `car` entity. As every row in the database represents a single `car`, the table itself holds multiple _cars_, hence I would use that as name, too.
Thinking about it, this is also more consistent with the principle, that each model should represent a single database table.